src/login/logind-core.c

```diff
@@ -817,3 +818,24 @@ void manager_reconnect_utmp(Manager *m) {
         manager_connect_utmp(m);
 #endif
 }
+
+int manager_read_efi_boot_loader_entries(Manager *m) {
+#if ENABLE_EFI
+        int r;
+
+        assert(m);
+        if (m->efi_boot_loader_entries_set)
+                return 0;
+
+        r = efi_loader_get_entries(&m->efi_boot_loader_entries);
+        if (IN_SET(r, -ENOENT, -EOPNOTSUPP))
+                return log_debug_errno(r, "Boot loader reported no entries.");
```

Comment:
Negative result (no entries) should be cached too.

Reply:
indeed